File: examples/sample.go
package main

import "fmt"

type Reader interface {
    Read(p []byte) (n int, err error)
    Close() error
}

type Writer interface {
    Write(p []byte) (n int, err error)
}

func fibonacci(n int) int {
    if n <= 1 {
        return n
    }
    return fibonacci(n-1) + fibonacci(n-2)
}

type Calculator struct {
    value int
}

type FileHandler struct {
    name string
}

func (c Calculator) Add(x, y int) int {
    return x + y
}

func (c Calculator) Multiply(x, y int) int {
    return x * y
}

func (f FileHandler) Read(p []byte) (n int, err error) {
    fmt.Println("Reading file:", f.name)
    return 0, nil
}

func (f FileHandler) Close() error {
    fmt.Println("Closing file:", f.name)
    return nil
}

func main() {
    callback := func(x int) int {
        return x * 2
    }
    result := callback(5)
}

File: examples/sample.rs
fn fibonacci(n: u32) -> u32 {
    if n <= 1 {
        return n;
    }
    fibonacci(n - 1) + fibonacci(n - 2)
}

pub struct Calculator {
    value: i32,
}

pub trait Arithmetic {
    fn add(&self, x: i32) -> i32;
    fn multiply(&self, x: i32) -> i32;
}

impl Calculator {
    pub fn new(initial: i32) -> Self {
        Calculator { value: initial }
    }

    pub fn add(&self, x: i32, y: i32) -> i32 {
        x + y
    }

    fn process(&self) {
        let closure = |x| x * 2;
        let result = closure(5);
    }
}

impl Arithmetic for Calculator {
    fn add(&self, x: i32) -> i32 {
        self.value + x
    }

    fn multiply(&self, x: i32) -> i32 {
        self.value * x
    }
}
